cpp/include/cuvs/neighbors/detail/ivf_common.cuh

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/linalg/unary_op.cuh>
```

Comment:
As mentioned before, all headers except for very basic vocabulary types that users should be interacting with, should be moved to `src/` and only those very public things should be defined in `include/`. Also, anything more than very trivial function implementations (like single-liners) should be compiled into source files and just defined in `include/` headers.